### PR TITLE
Create Task UI Implemented

### LINKED ISSUE #80: [Feature 11] make a "Create Task UI"
We have a UI for the TaskTree, but we don't have a UI for creating a Task. This needs to be implemented and connected to the appropriate use case.

This UI must have:
- A field for inputting the name of a Task
- A field for inputting the Task description
- A field for inputting the parent task's ID -- if the parent ID is invalid, the UI must notify the user and not make the task
- A button for creating the new task
- A button for canceling

The ID of the curriculum this task is put under is the ID of the TaskTree window, so this window is ID dependent.
If the name is empty, the creation is invalid, and the UI must notify the user.
This UI must reset its fields whenever the visibility is toggled

### PR DESCRIPTION
**Pull request by dadler83**
New Features:
- Create Task UI made and connected to Bailey's use case

Changes:
- The TaskDependentWindow and CurriculumDependentWindow interfaces have had their method names changed so they can be used in tandem

Testing: run the main at the bottom of the CreateTaskUI file; if you try to create a task, there will be a null pointer exception because the correct parent IDs have not been configured, and there is no set InMemoryUser, but everything is hooked up as needed.

### CODE REVIEW

**Review by MMachadoUofT — DISMISSED** [review]
This looks excellent! The implementation looks good, and if it seems to be hooked up properly with Baylee's use case, then it seems like were set. As I understand, Oswin was also looking at this use case, so when the time comes to hook it up to the leftover parts, we can delegate that work to him as needed. Nicely done, as usual!

**Review by Eggathin — DISMISSED** [review]
Allows the creation of new tasks though a user interface, which associates the new task with a parent task. Essential for the functionality of the project.

File: src/main/java/create_task_UI/CreateTaskUI.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    private void cancelButtonListener() {
+        this.cancelButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                programWindows.closeWindow(WindowManager.CREATE_TASK_REFERENCE_KEY);
+            }
+        });
+    }
+
+    // TODO: Delete for deployment
```
Comment (Eggathin) [inline src/main/java/create_task_UI/CreateTaskUI.java]: Ensure that this is deleted before deployment

**Review by bmguiler — DISMISSED** [review]
Looks good

**Review by bmguiler — APPROVED** [review]

**Review by MMachadoUofT — APPROVED** [review]
Glad you were able to resolve these conflicts! Hopefully this should work to merge now.

**Review by Eggathin — APPROVED** [review]
re-approval following removal of conflicts

**Pull request merged**